Sofa/framework/DefaultType/src/sofa/defaulttype/RigidDeriv.h

```diff
@@ -573,3 +575,38 @@ constexpr RigidDeriv<2,T> velocityAtRotatedPoint(const RigidDeriv<2,T>& v, const
 }
 
 } // namespace sofa::defaulttype
+
+
+namespace sofa::linearalgebra
+{
+    template <Size N, class T, typename IndexType>
+    class matrix_bloc_traits < defaulttype::RigidDeriv<N, T>, IndexType >
+    {
+    public:
+        typedef defaulttype::RigidDeriv<N, T> Block;
+        typedef T Real;
+        typedef Block BlockTranspose;
+
+        enum { NL = 1 };
+        enum { NC = defaulttype::RigidDeriv<N, T>::total_size };
+
+        static const Real& v(const Block& b, int /*row*/, int col) { return b[col]; }
+        static Real& v(Block& b, int /*row*/, int col) { return b[col]; }
+        static void vset(Block& b, int /*row*/, int col, Real v) { b[col] = v; }
+        static void vadd(Block& b, int /*row*/, int col, Real v) { b[col] += v; }
+        static void clear(Block& b) { b.clear(); }
+        static bool empty(const Block& b)
+        {
+            for (int i = 0; i < NC; ++i)
+                if (b[i] != 0) return false;
+            return true;
```

Comment:
Use `std::all_of` instead if possible

Reply:
not as straightforward : b is a RigidDeriv and as implemented its own `operator[]`, so it is not a real contihuous.
(https://github.com/sofa-framework/sofa/blob/3e107166c51e7a533ef29ad180ae9d7863e7bf3a/Sofa/framework/DefaultType/src/sofa/defaulttype/RigidDeriv.h#L226)
We would need to implement `begin()` and `end() `which is out the scope of the PR